pandas/core/dtypes/astype.py

```diff
@@ -125,6 +125,14 @@ def _astype_nansafe(
         )
         raise ValueError(msg)
 
+    if arr.dtype == object and dtype == bool:
+        # If the dtype is bool and the array is object, we need to replace
+        # the False and True of the object type in the ndarray with the
+        # bool type to ensure that the type conversion is correct
+        arr[arr == "False"] = np.False_
+        arr[arr == "True"] = np.True_
+        return arr.astype(dtype, copy=copy)
```

Comment:
I don't think we should be changing the behavior of `astype(bool)` to special case certain values.

Reply:
I got it. I think the method you mentioned in #60508  is quite reasonable. I'll make the modifications.